File: Algoritmos/C/Queue.c
//Para el otro método revisar la versión en java
#include<stdio.h>
#define SIZE 5
int values[SIZE], front = -1, rear= -1;


void enQueue(int value){
    if(rear == SIZE-1)
        printf("Nuestro Queue esta lleno\n" );
    else {
        if(front == -1)
            front = 0;
        rear++;
        values[rear] = value;
        printf("Se inserto el valor %d correctamente\n", value);
    
    }
 
}

void deQueue(){
    if(front == -1)
        printf("Nuestro Queue esta vacio\n" );
    else{
        printf("se elimino el valor %d\n", values[front]);
        front++;
        if(front > rear)
        front = rear = -1;
    }

}

main(int argc, char const *argv[])
{
    enQueue(1);
    enQueue(2);
    enQueue(3);
    enQueue(4);
    enQueue(5);
    deQueue();
    deQueue();
    deQueue();
    deQueue();
    deQueue();
    enQueue(10);
    return 0;
}

File: Algoritmos/Java/Queue.java
class Queue {
  public static int size;
  public static int front = -1;
  public static int rear =-1;
  public static int x;
  public static int[] items;
  public static void main(final String[] args) throws Exception{
    items = new int[5];
    size = items.length;
    //Aquí en adelante puedes ingresar tus acciones
    enQueue(5);
    enQueue(6);
    deQueue();
    deQueue();
    deQueue();
    enQueue(1);
    enQueue(2);
    enQueue(3);
    enQueue(4);
    enQueue(5);
    enQueue(6);
    deQueue();
    deQueue();
    deQueue();
    enQueue(6);
    enQueue(7);
    enQueue(8);
    printDataOrder();
  }

  public static void enQueue(Integer value)  throws Exception{
    if(rear == size-1)System.out.println("La cola esta llena");
    else {
      if(front == -1)
        front = 0;
      rear++;
      items[rear] = value;
      System.out.println("El valor "+value+" fue agregado correctamente");
    }
  }
  public static void deQueue() throws Exception{
    if(rear == -1){
      System.out.println("Nuestra Queue esta vacia");
    }
    else{
      System.out.println("Fue eliminado el valor: "+items[0]);
      for(x=0; x<rear; x++){
        items[x]=items[x+1];
      }
      items[rear] = 0;
      rear--;
      if(rear == -1) front = rear;
    }
  }
  public static void printDataOrder() throws Exception{
    for(x=0; x<size; x++){
      System.out.print(items[x]+" ");
    }
    System.out.println();
  }
}